core/serialization/i_plugin_serializer.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The record type for an object containing functions for
+ *     serializing part of the workspace.
+ */
+
+'use strict';
+
+goog.module('Blockly.serialization.IPluginSerializer');
+goog.module.declareLegacyNamespace();
+
+// eslint-disable-next-line no-unused-vars
+const Workspace = goog.requireType('Blockly.Workspace');
+
+
+/**
+ * Serializes and deserializes a plugin.
+ * @record
+ */
+class IPluginSerializer {
+  constructor() {
+    /**
+     * A priority value used to determine the order of deserializing plugins.
+     * More positive priorities are deserialized before less positive
+     * priorities. Eg if you have priorities (0, -10, 10, 100) the order of
+     * deserialiation will be (100, 10, 0, -10).
+     * If two plugins have the same priority, they are deserialized in an
+     * arbitrary order relative to eachother.
+     * @type {number}
+     */
+    this.priority;
+  }
+
+  /* eslint-disable no-unused-vars, valid-jsdoc */
+
+  /**
+   * Saves the state of the plugin.
+   * @param {!Workspace} workspace The workspace the plugin to serialize is
+   *     associated with.
+   * @return {?Object} A JS object containing the plugin's state, or null if
+   *     there is no state to record.
+   */
+  save(workspace) { }
```

Comment:
I think since this is a record you don't need the `{}`

Reply:
Hmm it looks like you need them for non-static methods, but not static methods ([5.4.9](https://google.github.io/styleguide/jsguide.html#features-classes-interfaces)) And also there's not supposed to be an extra space, so thanks for the catch :D